src/routers/router.test.ts

```diff
@@ -0,0 +1,70 @@
+import { getApp } from '../servers/server'
+import request from 'supertest'
+
+const app = getApp();
+
+describe('POST', () => {
+    it('POST success', async () => {
+        const response = await request(app.callback()).post('/').send({
+            data: 'something',
+        })
+        expect(response.status).toEqual(200)
+        expect(response.text).toEqual('Hello World')
+
+    })
+})
+
+describe('GET', () => {
+    it('GET success', async () => {
+        const response = await request(app.callback()).get('/')
+        expect(response.status).toEqual(200)
+        expect(response.text).toEqual('Hello World')
+    })
+})
+
+describe('PUT', () => {
+    it('PUT success', async () => {
+        const response = await request(app.callback()).put('/').send({
+            data: 'somthing',
+        })
+        expect(response.status).toEqual(200)
+        expect(response.text).toEqual('Hello World')
+
+    })
+})
+
+describe('DELETE', () => {
+
+    it('DELETE user not found', async () => {
+        const response = await request(app.callback()).delete('/0');
+        expect(response.status).toEqual(404);

```

Comment:
come mai esce 404?

Reply:
perché nell'array non ci sono utente con id=0, quindi mi da not found

Reply:
perché nell'array non ci sono utenti con id=0, quindi users.findIndex(...) ritorna -1 e non entra nell'if in cui in cui assegno il valore a ctx.body, cioè io almeno avevo interpretato così